l337-postgres/src/lib.rs

```diff
@@ -144,154 +141,127 @@ where
+            let rows = conn.client.query(&select, &[]).await.unwrap();
+
+            for row in rows {
+                assert_eq!(2, row.get(0));
+            }
 
-        runtime.block_on(future).expect("could not run");
+            delay_for(Duration::from_secs(5)).await;
+
+            conn
+        };
+
+        futures::join!(q1, q2);
     }
 
-    #[test]
-    fn it_reuses_connections() {
+    #[tokio::test]
+    async fn it_reuses_connections() {
         let mngr = PostgresConnectionManager::new(
             "postgres://pass_user:password@localhost:5433/postgres"
                 .parse()
                 .unwrap(),
             tokio_postgres::NoTls,
         );
 
-        let mut runtime = Runtime::new().expect("could not run");
         let config: Config = Default::default();
-        let future = Pool::new(mngr, config).and_then(|pool| {
-            let q1 = pool.connection().and_then(|mut conn| {
-                conn.client
-                    .prepare("SELECT 1::INT4")
-                    .and_then(move |select| {
-                        conn.client.query(&select, &[]).for_each(|row| {
-                            assert_eq!(1, row.get::<_, i32>(0));
-                            Ok(())
-                        })
-                    })
-                    .map(|connection| {
-                        sleep(Duration::from_secs(5));
-                        ((), connection)
-                    })
-                    .map_err(|e| l337::Error::External(e))
-            });
-
-            let q2 = pool.connection().and_then(|mut conn| {
-                conn.client
-                    .prepare("SELECT 2::INT4")
-                    .and_then(move |select| {
-                        conn.client.query(&select, &[]).for_each(|row| {
-                            assert_eq!(2, row.get::<_, i32>(0));
-                            Ok(())
-                        })
-                    })
-                    .map(|connection| {
-                        sleep(Duration::from_secs(5));
-                        ((), connection)
-                    })
-                    .map_err(|e| l337::Error::External(e))
-            });
-
-            let q3 = pool.connection().and_then(|mut conn| {
-                conn.client
-                    .prepare("SELECT 3::INT4")
-                    .and_then(move |select| {
-                        conn.client.query(&select, &[]).for_each(|row| {
-                            assert_eq!(3, row.get::<_, i32>(0));
-                            Ok(())
-                        })
-                    })
-                    .map(|connection| {
-                        sleep(Duration::from_secs(5));
-                        ((), connection)
-                    })
-                    .map_err(|e| l337::Error::External(e))
-            });
-
-            q1.join3(q2, q3)
-        });
+        let pool = Pool::new(mngr, config).await.unwrap();
```

Comment:
Do you know how this test proves that it reuses connections? I don't see anything functionally different from the previous implementation so this change is fine, just confused on what it was doing before

Reply:
Looking at it, I don't think this test proves that it reuses connections at all